gen/weather/weather.hoon

```diff
@@ -0,0 +1,52 @@
+/-  sole
+/+  generators
+=,  generators
+:-  %get
+|=  [* [city=tape country=tape appid=tape ~] ~]
+^-  (sole-request:sole (cask tang))
+=/  escaped-city  (en-urlt:html city)
+=/  escaped-country  (en-urlt:html country)
+%+  curl  (scan "http://api.openweathermap.org/data/2.5/weather?q={escaped-city}%2C{escaped-country}&APPID={appid}" auri:de-purl:html)
+|=  hit/httr:eyre
+?~  r.hit  !!
+=/  my-json  (de-json:html q.u.r.hit)
+?~  my-json  !!
+=,  dejs:format
+=/  my-map
+%-  (om same)
+u.my-json
+=/  name
+%.  %name
+%~  got  by
+my-map
```

Comment:
You can just write these as `(~(got by my-map) %name)`. It's short enough for a single line and reads a little easier than the tall form version.